### PR TITLE
APB-7583 (iw) add session id and request id to off-platform requests

### PR DESCRIPTION
**Pull request by ian-1**

### CODE REVIEW

File: app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala [hunk 1]
```diff
@@ -302,10 +304,12 @@ class IFConnector @Inject()(httpClient: HttpClient, metrics: Metrics, agentCache
         ec)
     }
 
-  private def ifHeaders(authToken: String, env: String) = Seq(
+  def ifHeaders(authToken: String, env: String)(implicit hc: HeaderCarrier) = Seq(
     HeaderNames.authorisation -> s"Bearer $authToken",
     Environment               -> env,
-    CorrelationId             -> UUID.randomUUID().toString
+    CorrelationId             -> UUID.randomUUID().toString,
+    SessionId                 -> hc.sessionId.map(_.value).getOrElse(""),
```
Comment (gitwojciech) [inline app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:311]: Do not know spec but why we add headers with empty strings ?

Comment (ian-1) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:311]: The empty string was only as a default if no sessionId found. Might not be the best way to deal with it?

Comment (gitwojciech) [reply app/uk/gov/hmrc/agentclientrelationships/connectors/IFConnector.scala:311]: I thiink when no session not found we should not send that header.

**Review by CarolineS-QA — APPROVED** [review]

File: app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala [hunk 2]
```diff
@@ -111,11 +113,14 @@ class DesConnector @Inject() (httpClient: HttpClient, metrics: Metrics, agentCac
...
-    Environment               -> env,
-    CorrelationId             -> UUID.randomUUID().toString
-  )
+  def desHeaders(authToken: String, env: String)(implicit hc: HeaderCarrier): Seq[(String, String)] =
+    Seq(
+      HeaderNames.authorisation -> s"Bearer $authToken",
+      Environment               -> env,
+      CorrelationId             -> UUID.randomUUID().toString
+    ) ++
+      (if (hc.sessionId.map(_.value).nonEmpty) Seq(SessionId -> hc.sessionId.map(_.value).get) else Seq.empty) ++
```
Comment (gitwojciech) [inline app/uk/gov/hmrc/agentclientrelationships/connectors/DesConnector.scala:122]: Could you please use map or fold instead ?? hc.sessionId.fold(Seq.empty[(String,String)])(x=>Seq(SessionId -> x.value)) or with map would be hc.sessionId.map(x=> Seq(SessionId -> x.value)).getOrElse(Seq.empty[(String,String)])

**Pull request merged**